### ISSUE TITLE
new lint: find length comparisons after access by index in ANDded if-chains

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#5785]
### What it does
Check for code like
````rust
fn a(v1: Vec<_>, v2: Vec<_>, x: usize) {
  if v1[x] == v2[x] &&
     v1.len() == v2.len() {}
}
````
The length-check is probably intended to guard against index-out-of-bounds access however it was (most likely mistakenly) placed AFTER the access by index. It should be at the front. 

### Categories (optional)

- Kind: correctness

Inspired by https://github.com/rust-lang/rust-clippy/issues/5780 and https://github.com/rust-lang/rust-clippy/pull/5784

**Comment by GuillaumeGomez** [comment]
I would even suggest to use `get()` instead. :)
